lib/Onyx.js

```diff
@@ -496,22 +486,23 @@ function applyMerge(key, data) {
  *
  * @param {string} key
  * @param {*} val
+ * @returns {Promise}
```

Comment:
On the other hand, the worst part for **not** returning a promise is the client of Onyx won't have a clue whether `merge` succeed or failed. This might be intended for E.cash, but I as a user would like to be free to handle the promise

Reply:
> as a library user would like to be free to handle the promise

I think in this case, we would rather be opinionated about how Onyx is utilized, rather than just allowing for a lot of "use it however you want" scenarios. It doesn't feel like this is the right PR to be focused on how a consumer other than e.cash should behave. So, just to be clear, I'm not saying that we shouldn't have that convo, I just don't think this is the right place to have it.

If there is a way to get around not needing to return this Promise, I think we should do it. Though, if it's absolutely required, then I guess I would rather return a promise to have the benchmarking as opposed to nothing.

Reply:
Also, I see that we return promises from everything except `merge()`, so that weakens my argument tremendously already, and I don't think returning a promise here will have any significant impact.